src/metabase/driver/sql/query_processor.clj

```diff
@@ -1260,6 +1260,10 @@
   [driver [_ value unit]]
   (date driver unit (->honeysql driver value)))
 
+(defmethod ->honeysql [:sql :date]
```

Comment:
If the method is implemented here now, should the `:expressions/date` feature also get enabled on the `:sql` driver? I'm actually not sure what the answer is, but intuitively that's what I'd expect.

Reply:
Yes. That's happening in another branch.